### ISSUE TITLE
Wavpack DSD files show wrong duration

### ISSUE THREAD

**Issue by the-black-eagle** [taglib/taglib#962]
When attempting to get the duration of a wavpack DSD file, taglib returns a duration that is roughly 4 times what it should be. For instance, for the Genesis track Hairless Heart, the duration should be 2:09 or thereabouts.  Taglib returns a duration of 8:41 for this track. 
This issue was first raised on the Kodi forum (https://forum.kodi.tv/showthread.php?tid=355493) as Kodi uses taglib for the majority of it's tag reading.  I would therefore expect that this could impact a large number of users given that Kodi can now play these files.

I can supply a test file if required.

**Comment by dbry** [comment]
I took a quick look at the taglib WavPack code and the problem is that it only seems to examine the 32-byte WavPack header. This is sufficient for most things, but there are a couple cases where a little more work needs to be done to determine the actual sample rate of the file (and it's the sample rate that's wrong here, which of course affects the reported duration).

The first case is when a "non-standard" sampling rate is used and the other is for DSD audio. I recently provided a [patch for MKVToolNix](https://gitlab.com/mbunkus/mkvtoolnix/-/merge_requests/2204) for this exact issue (plus another issue involving checksums) and all the information and code is there.

I'll see if I can get a chance to create a patch for taglib to do this. As I mentioned in the other thread I'm not a strong C++ programmer, so hopefully someone else can carefully review it. Or, there's enough code in the other patch to allow someone else to do this, but I'm thinking that the WavPack code in taglib is old enough that nobody is familiar with it any more.

Thanks!
